### PR TITLE
Limit query concurrency in salesforce

### PR DESCRIPTION
**Pull request by ori-moisis**
Limiting the number of concurrent query requests seems to have a significant
positive impact on query performance so in addition to adding the configuration
option we also set a default limit of 4

---

Noticed this while fixing [SALTO-1274](https://salto-io.atlassian.net/browse/SALTO-1274), with concurrency capped at 4 fetching ~1500 instances took ~5 seconds and with unlimited concurrency it could take up to a minute.

---
_Release Notes_: 
Salesforce Adapter
- Added configuration option to limit the maximum concurrency of query requests

### CODE REVIEW

**Review by netama — APPROVED** [review]
LGTM.
did you also check the effect of the limit on the dependency (tooling) API queries? any chance we want different limits for tooling and for the regular api (are they even using the same resources)?

File: packages/salesforce-adapter/src/client/client.ts [hunk 1]
```diff
@@ -497,11 +498,26 @@ export default class SalesforceClient {
     )
   }
 
+  @throttle<ClientRateLimitConfig>('query')
```
Comment (netama) [inline packages/salesforce-adapter/src/client/client.ts:501]: nit: could be nice to include some keys in the throttle log to make it easier to know what was queried when. but not sure we have a good way without printing the whole query and I guess that might be too long?

Comment (ori-moisis) [reply packages/salesforce-adapter/src/client/client.ts:501]: currently the query is printed to the log

**Comment by ori-moisis** [comment]
I did not check the effect on tooling, unclear if they affect each other, but with the current code this won't actually be constrained by this new limit because filters do not run concurrently.

**Comment by netama** [comment]
the extra-dependencies filter itself has concurrent `queryAll` calls, so they will be throttled. but probably not a major issue, right now we have 6

**Pull request merged**